### PR TITLE
Charlesmchen/unread indicator appearance2

### PR DESCRIPTION
**Pull request by charlesmchen**
### This PR is now ready for review.

This PR reworks the appearance of the unread indicator.

PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
This PR is now ready for review.

PTAL @michaelkirk

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 1]
```diff
@@ -109,6 +110,40 @@
...
+@end
+
+#pragma mark -
+
+@implementation OWSMessagesCollectionViewFlowLayout
+
+- (CGSize)sizeForItemAtIndexPath:(NSIndexPath *)indexPath
+{
+    // The unread indicator should be sized according to its desired size.
+    if ([self.delegate isSpecialItemAtIndexPath:indexPath]) {
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: In subsequent PRs, we use this same technique to do custom sizing of the system messages.

**Comment by charlesmchen** [comment]
### Before

screen shot 2017-06-05 at 4 29 20 pm

### After

screen shot 2017-06-05 at 4 28 34 pm

I still like the old design better.

**Comment by michaelkirk** [comment]
> I still like the old design better.

Yeah, me too.

The code LGTM.

**Pull request merged**